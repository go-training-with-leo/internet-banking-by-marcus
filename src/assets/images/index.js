import { ReactComponent as PlusIcon } from 'assets/images/icons/plus.svg';
import { ReactComponent as CreditCard } from 'assets/images/icons/creditCard.svg';
import { ReactComponent as Contact } from 'assets/images/icons/contact.svg';
import { ReactComponent as CreditCardDone } from 'assets/images/icons/creditCardDone.svg';
import { ReactComponent as Clock } from 'assets/images/icons/time.svg';
import { ReactComponent as DeleteIcon } from 'assets/images/icons/delete.svg';
import { ReactComponent as Edit } from 'assets/images/icons/edit.svg';
import { ReactComponent as Transfer } from 'assets/images/icons/transfer.svg';
import { ReactComponent as Cash } from 'assets/images/icons/cash.svg';
import { ReactComponent as CashAdd } from 'assets/images/icons/cashAdd.svg';
import { ReactComponent as Cancel } from 'assets/images/icons/cancel.svg';
import { ReactComponent as User } from 'assets/images/icons/user.svg';
import { ReactComponent as ArrowDown } from 'assets/images/icons/arrowDown.svg';
import { ReactComponent as Check } from 'assets/images/icons/check.svg';
import { ReactComponent as Reverse } from 'assets/images/icons/reverse.svg';
import { ReactComponent as NotifyBusy } from 'assets/images/icons/notificationBusy.svg';
import { ReactComponent as Info } from 'assets/images/icons/info.svg';
import { ReactComponent as NotifyFree } from 'assets/images/icons/notificationFree.svg';
import { ReactComponent as NotifyRemove } from 'assets/images/icons/notificationRemove.svg';
import { ReactComponent as Alert } from 'assets/images/icons/alert.svg';
import { ReactComponent as AlertYellow } from 'assets/images/icons/alertYellow.svg';
import { ReactComponent as Locker } from 'assets/images/icons/locker.svg';
import { ReactComponent as EightLogo } from 'assets/images/icons/eightBank.svg';
import { ReactComponent as MasterCard } from 'assets/images/icons/masterCard.svg';
import { ReactComponent as Next } from 'assets/images/icons/next.svg';
import { ReactComponent as Back } from 'assets/images/icons/back.svg';
import { ReactComponent as Ellipse } from 'assets/images/icons/ellipse.svg';
import { ReactComponent as Search } from 'assets/images/icons/search.svg';

import { ReactComponent as Loading } from 'assets/images/icons/loadingV2.svg';
import { ReactComponent as Visa } from 'assets/images/icons/visa.svg';
import { ReactComponent as Napas } from 'assets/images/icons/napas.svg';

import Eye from 'assets/images/icons/eye.png';
import EyeCross from 'assets/images/icons/eyeCross.png';

import { ReactComponent as ACB } from 'assets/images/icons/acb.svg';

import EightGif from './eight.gif';

export {
  ACB,
  Alert,
  AlertYellow,
  ArrowDown,
  Back,
  Cancel,
  Cash,
  CashAdd,
  Check,
  Clock,
  Contact,
  CreditCard,
  CreditCardDone,
  DeleteIcon,
  Edit,
  EightGif,
  EightLogo,
  Ellipse,
  Eye,
  EyeCross,
  Info,
  Loading,
  Locker,
  MasterCard,
  Napas,
  Next,
  NotifyBusy,
  NotifyFree,
  NotifyRemove,
  PlusIcon,
  Reverse,
  Search,
  Transfer,
  User,
  Visa,
};
