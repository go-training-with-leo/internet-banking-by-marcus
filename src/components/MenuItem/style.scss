@import "assets/scss/mixins.scss";
@import "assets/scss/variables.scss";
@import "assets/scss/extends.scss";

.menu-item {
  @include setWidthHeight(197px, 67px);
  @include setFlex(row, start, center, 22px, null);
  padding-left: 26px;
  background-color: $black-medium;
  color: $pure-white;
  
  font-family: $font-family;
  font-weight: 700;
  cursor: pointer;
  transition: all 0.4s ease-in-out;

  &:hover {
    background-image: $menu-gradient;
  }
}
