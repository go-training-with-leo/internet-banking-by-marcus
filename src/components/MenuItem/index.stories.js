import React from 'react';

import MenuItem from 'components/MenuItem';

import { PlusIcon } from 'assets/images';

export default {
  title: 'Menu Item',
  component: MenuItem,
};

const Template = (args) => {
  return <MenuItem {...args} />;
};

const Default = Template.bind({});

Default.args = {
  icon: <PlusIcon fill='gray' />,
  children: 'Label',
};

export { Default };
