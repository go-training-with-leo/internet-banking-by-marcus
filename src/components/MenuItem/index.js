import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import 'components/MenuItem/style.scss';

const MenuItem = ({ icon, isActive, children }) => {
  return (
    <div
      className={classNames('menu-item', {
        active: isActive,
      })}
    >
      {icon}
      {children}
    </div>
  );
};

MenuItem.defaultProps = {
  isActive: false,
  icon: undefined,
  children: undefined,
};

MenuItem.propTypes = {
  isActive: PropTypes.bool,
  icon: PropTypes.node,
  children: PropTypes.string,
};

export default MenuItem;
