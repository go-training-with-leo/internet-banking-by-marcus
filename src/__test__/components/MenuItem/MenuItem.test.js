import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import MenuItem from 'components/MenuItem';
import { PlusIcon } from 'assets/images';

test('Test default props menu item', () => {
  const tree = render(<MenuItem />);

  const menuItem = tree.getByRole('listitem');

  expect(menuItem.className).toMatch('menu-item');

  expect(menuItem.onclick).toEqual(null);

  expect(tree).toMatchSnapshot();
});

test('Test all props menu item', () => {
  const handleClick = jest.fn();

  const tree = render(
    <MenuItem icon={<PlusIcon />} onClick={handleClick}>
      Label
    </MenuItem>
  );

  const menuItem = tree.getByRole('listitem');
  const icon = menuItem.firstChild;
  fireEvent.click(menuItem);
  fireEvent.mouseMove(menuItem);
  expect(icon).toHaveAttribute('fill', '#FFF');
  fireEvent.mouseLeave(menuItem);

  expect(handleClick).toHaveBeenCalled();
  expect(menuItem.className).toMatch('menu-item');

  expect(tree).toMatchSnapshot();
});

test('Test all props menu item with ', () => {
  const handleClick = jest.fn();

  const tree = render(
    <MenuItem icon={<PlusIcon />} isActive onClick={handleClick}>
      Label
    </MenuItem>
  );

  const menuItem = tree.getByRole('listitem');
  const icon = menuItem.firstChild;
  fireEvent.click(menuItem);
  fireEvent.mouseMove(menuItem);
  expect(icon).toHaveAttribute('fill', '#EF230C');
  fireEvent.mouseLeave(menuItem);

  expect(handleClick).toHaveBeenCalled();
  expect(menuItem.className).toMatch('menu-item');

  expect(tree).toMatchSnapshot();
});
