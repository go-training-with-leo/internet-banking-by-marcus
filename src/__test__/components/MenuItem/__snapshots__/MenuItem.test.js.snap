// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test all props menu item 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="menu-item"
        role="listitem"
      >
        <svg>
          Cross.svg
        </svg>
        Label
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="menu-item"
      role="listitem"
    >
      <svg>
        Cross.svg
      </svg>
      Label
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test all props menu item with  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="menu-item active"
        role="listitem"
      >
        <svg
          fill="#EF230C"
        >
          Cross.svg
        </svg>
        Label
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="menu-item active"
      role="listitem"
    >
      <svg
        fill="#EF230C"
      >
        Cross.svg
      </svg>
      Label
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test default props menu item 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="menu-item"
        role="listitem"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="menu-item"
      role="listitem"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
